stl/inc/vector

```diff
@@ -458,9 +492,116 @@ public:
+                return;
+            }
+
+            const auto& _My_data = _Target_vec._Mypair._Myval2;
+            _Asan_vector_aligned_annotate_start(_Unfancy_maybe_null(_My_data._Myfirst),
+                _Unfancy_maybe_null(_My_data._Mylast), _Unfancy_maybe_null(_My_data._Myend), _New_size,
+                sizeof(value_type), &_Asan_last);
+        }
+
+        _CONSTEXPR20 ~_Asan_annotater() {
+#if _HAS_CXX20
+            if (_STD is_constant_evaluated()) {
+                return;
+            }
+#endif // _HAS_CXX20
+            if (!_Asan_vector_should_annotate) {
+                return;
+            }
+
+            const auto& _My_data = _Target_vec._Mypair._Myval2;
+            _Asan_vector_aligned_annotate_end(_Unfancy_maybe_null(_My_data._Myfirst),
+                _Unfancy_maybe_null(_My_data._Mylast), _Unfancy_maybe_null(_My_data._Myend), _Asan_last);
+        }
+
+    private:
+        const vector& _Target_vec;
+        const void* _Asan_last;
+    };
+
+    template <>
+    class _NODISCARD _Asan_annotater<false> {
+    public:
+        _Asan_annotater(const _Asan_annotater&) = delete;
+        _Asan_annotater& operator=(const _Asan_annotater&) = delete;
+
+        explicit constexpr _Asan_annotater(const vector& _Target) noexcept
+            : _Target_vec(_Target), _Asan_last(nullptr), _Aligned_first(nullptr) {}
+
+        _CONSTEXPR20 _Asan_annotater(const vector& _Target, size_type _New_size)
+            : _Target_vec(_Target), _Asan_last(nullptr) {
+#if _HAS_CXX20
+            if (_STD is_constant_evaluated()) {
+                return;
+            }
+#endif // _HAS_CXX20
+            if (!_Asan_vector_should_annotate) {
+                return;
+            }
+
+            const auto& _My_data = _Target_vec._Mypair._Myval2;
+            _Asan_vector_unaligned_annotate_start(_Unfancy_maybe_null(_My_data._Myfirst),
+                _Unfancy_maybe_null(_My_data._Mylast), _Unfancy_maybe_null(_My_data._Myend), _New_size,
+                sizeof(value_type), &_Asan_last, &_Aligned_first);
+        }
+
+        _CONSTEXPR20 ~_Asan_annotater() {
+#if _HAS_CXX20
+            if (_STD is_constant_evaluated()) {
+                return;
+            }
+#endif // _HAS_CXX20
+            if (!_Asan_vector_should_annotate) {
+                return;
+            }
+
+            const auto& _My_data = _Target_vec._Mypair._Myval2;
+            _Asan_vector_unaligned_annotate_end(_Unfancy_maybe_null(_My_data._Myfirst),
+                _Unfancy_maybe_null(_My_data._Mylast), _Unfancy_maybe_null(_My_data._Myend), _Asan_last,
+                _Aligned_first);
+        }
+
+    private:
+        const vector& _Target_vec;
+        const void* _Asan_last;
+        const void* _Aligned_first;
+    };
+
     using _Scary_val = _Vector_val<conditional_t<_Is_simple_alloc_v<_Alty>, _Simple_types<_Ty>,
         _Vec_iter_types<_Ty, size_type, difference_type, pointer, const_pointer, _Ty&, const _Ty&>>>;
 
+    template <class... _Valty>

```

Comment:
I am not really a fan of this drive by change

Reply:
It was changed to extract the functionality out of `emplace_back` in order to make annotating the input_iterator insert range more efficient.